src/openms/source/FORMAT/DATAACCESS/SiriusMzTabWriter.cpp

```diff
@@ -72,28 +72,40 @@ String SiriusMzTabWriter::extractFeatureId(const String& path)
   return feature_id;
 }
 
+std::map< String, Size > SiriusMzTabWriter::extract_columnname_to_columnindex(CsvFile& csvfile)
```

Comment:
This file share A LOT with CSIFingerIDMzTabWriter.
In general, why can't it use the general MzTabWriter? No small molecule support? Or is there a SMMZTabWriter?

Reply:
Yes, there are some overlapping points. Most of the overlap is refactored (I extracted an additional function - next push), but SiriusMzTabWriter will read `"/formula_candidates.tsv"`, which is structurally different from the CSIFingerID 
`"/structure_candidates.tsv"`. That why I have two individual readers/writers. 

The overlapping methods are implemented in the SiriusMzTabWriter and then used in the CSI one
```
   /**
    @brief Extract scan_index from filepath
    */
    static int extractScanIndex(const String& path);

    /**
    @brief Extract scan_number from filepath
    */
    static int extractScanNumber(const String& path);

    /**
    @brief Extract feature_id from filepath
    */
    static String extractFeatureId(const String& path);

    /**
    @brief Extract columnname and index based in SIRIUS entries
    */
    static std::map< String, Size > extract_columnname_to_columnindex(CsvFile& csvfile);

    /**
     @brief Extract mz, rt of the precursor and the nativeID of the corresponding MS2 spectra in the spectrum.ms file
    */
    static SiriusSpectrumMSInfo extractSpectrumMSInfo(const String& single_sirius_path);
```